msticpy/nbtools/timeline_duration.py

```diff
@@ -0,0 +1,248 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Timeline duration control."""
+from datetime import datetime
+from typing import Iterable, List, Optional, Tuple, Union
+
+import attr
+import pandas as pd
+from bokeh.io import output_notebook, show
+from bokeh.models import ColumnDataSource, HoverTool, LayoutDOM
+
+# pylint: enable=no-name-in-module
+from bokeh.plotting import figure, reset_output
+from bokeh.layouts import column
+from bokeh.transform import dodge
+
+from ..common.utility import check_kwargs
+from .timeline import (
+    _create_tool_tips,
+    _create_range_tool,
+    _calc_auto_plot_height,
+    _get_time_bounds,
+    _set_axes_and_grids,
+    _get_tick_formatter,
+    _plot_ref_events,
+)
+
+# pylint: disable=unused-import
+# Importing to activate pandas accessors
+from .timeline_pd_accessor import TimeLineAccessor  # noqa F401
+
+# pylint: enable=unused-import
+
+from .._version import VERSION
+
+__version__ = VERSION
+__author__ = "Ian Hellen"
+
+
+@attr.s(auto_attribs=True)
+class PlotParams:
+    """Plot params for time_duration."""
+
+    height: Optional[int] = None
+    width: int = 900
+    title: Optional[str] = None
+    yaxis: bool = True
+    range_tool: bool = True
+    xgrid: bool = True
+    ygrid: bool = False
+    hide: bool = False
+    color: str = "navy"
+    ylabel_cols: Iterable[str] = attr.Factory(list)
+    ref_events: Optional[pd.DataFrame] = None
+    ref_col: Optional[str] = None
+    ref_times: Optional[List[Tuple[datetime, str]]] = None
+
```

Comment:
`range_tool` and `source_columns` are missing :)

Reply:
Thx. source_columns is missing by design. Since the way I create the durations is by grouping on the group_by column(s) and then finding the max and min of the timestamps, all of the other field values for individual events disappear. 
They kind of have to be ignored really since I'm trying to assign Tooltip values to the entire duration - so only values that are common to the grouped duration can be shown.

range_tool is an omission. I think I've fixed in an update with tests that I'm about to push.